components/apimgt/org.wso2.carbon.apimgt.gateway/src/test/java/org/wso2/carbon/apimgt/gateway/inbound/websocket/request/GraphQLRequestProcessorTest.java

```diff
@@ -315,6 +322,59 @@ public void testHandleRequestInvalidScope() throws Exception  {
         Assert.assertFalse(processorResponseDTO.isCloseConnection());
     }
 
+    @Test
+    public void testHandleRequestScopeValidationSkipWhenSecurityDisabled() throws Exception  {
+
+        InboundMessageContext inboundMessageContext = new InboundMessageContext();
+        int msgSize = 100;
+        String msgText = "{\"id\":\"1\",\"type\":\"start\",\"payload\":{\"variables\":{},\"extensions\":{},"
+                + "\"operationName\":null,\"query\":\"subscription {\\n  "
+                + "liftStatusChange {\\n    id\\n    name\\n }\\n}\\n\"}}";
+        PowerMockito.mockStatic(InboundWebsocketProcessorUtil.class);
+        InboundProcessorResponseDTO responseDTO = new InboundProcessorResponseDTO();
+        PowerMockito.when(InboundWebsocketProcessorUtil.authenticateToken(inboundMessageContext))
+                .thenReturn(responseDTO);
+
+        // Get schema and parse
+        String graphqlDirPath = "graphQL" + File.separator;
+        String relativePath = graphqlDirPath + "schema_with_additional_props.graphql";
+        String schemaString = IOUtils.toString(getClass().getClassLoader().getResourceAsStream(relativePath));
+        SchemaParser schemaParser = new SchemaParser();
+        TypeDefinitionRegistry registry = schemaParser.parse(schemaString);
+        GraphQLSchema schema = UnExecutableSchemaGenerator.makeUnExecutableSchema(registry);
+        GraphQLSchemaDTO schemaDTO = new GraphQLSchemaDTO(schema, registry);
+        inboundMessageContext.setGraphQLSchemaDTO(schemaDTO);
+
+        // VerbInfoDTO with security disabled
+        VerbInfoDTO verbInfoDTO = new VerbInfoDTO();
+        verbInfoDTO.setHttpVerb("SUBSCRIPTION");
+        verbInfoDTO.setThrottling("Unlimited");
+        verbInfoDTO.setAuthType("None");
+        PowerMockito.when(InboundWebsocketProcessorUtil.findMatchingVerb("liftStatusChange",
+                inboundMessageContext)).thenReturn(verbInfoDTO);
+
+        // Creating response for scope validation
+        GraphQLProcessorResponseDTO graphQLProcessorResponseDTO = new GraphQLProcessorResponseDTO();
```

Comment:
Since this test should check skipping the validation of scopes when the security is disabled, we should write the test case similar to the **testHandleRequestSuccess** because, from this point onwards, we should check for successful invocation. (The test case should change to something like below from here onwards)

```
PowerMockito.when(InboundWebsocketProcessorUtil.findMatchingVerb("liftStatusChange", inboundMessageContext))
                .thenReturn(verbInfoDTO);
        APIKeyValidationInfoDTO infoDTO = new APIKeyValidationInfoDTO();
        infoDTO.setGraphQLMaxComplexity(4);
        infoDTO.setGraphQLMaxDepth(3);
        inboundMessageContext.setInfoDTO(infoDTO);
        PowerMockito.when(InboundWebsocketProcessorUtil.doThrottleForGraphQL(msgSize, verbInfoDTO,
                inboundMessageContext, "1")).thenReturn(responseDTO);
        GraphQLRequestProcessor graphQLRequestProcessor = new GraphQLRequestProcessor();
        InboundProcessorResponseDTO processorResponseDTO =
                graphQLRequestProcessor.handleRequest(msgSize, msgText, inboundMessageContext);
        Assert.assertFalse(processorResponseDTO.isError());
        Assert.assertNull(processorResponseDTO.getErrorMessage());
        Assert.assertEquals(inboundMessageContext.getVerbInfoForGraphQLMsgId("1").getOperation(), "liftStatusChange");
        Assert.assertEquals(inboundMessageContext.getVerbInfoForGraphQLMsgId("1").getVerbInfoDTO().getHttpVerb(),
                "SUBSCRIPTION");
        Assert.assertEquals(inboundMessageContext.getVerbInfoForGraphQLMsgId("1").getVerbInfoDTO().getThrottling(),
                "Unlimited");
Assert.assertEquals(inboundMessageContext.getVerbInfoForGraphQLMsgId("1").getVerbInfoDTO().getAuthType(),
                "None");
```

Reply:
graphQLProcessorResponseDTO is created to respond to the desired object if the scope is validated. But. here, we did not receive the expected graphQLProcessorResponseDTO, as the actual flow avoids validation of the scope due to disabled security in the requested operation.

```
PowerMockito.when(InboundWebsocketProcessorUtil.validateScopes(inboundMessageContext,
                "liftStatusChange", "1")).thenReturn(graphQLProcessorResponseDTO);
```

Reply:
Ack. Thanks